src/checkout-buttons/create-checkout-button-registry.ts

```diff
@@ -0,0 +1,43 @@
+import { createFormPoster } from '@bigcommerce/form-poster';
+import { createRequestSender } from '@bigcommerce/request-sender';
+import { getScriptLoader } from '@bigcommerce/script-loader';
+
+import { CheckoutActionCreator, CheckoutRequestSender, CheckoutStore } from '../checkout';
+import { Registry } from '../common/registry';
+import { ConfigActionCreator, ConfigRequestSender } from '../config';
+import { BraintreeScriptLoader, BraintreeSDKCreator } from '../payment/strategies/braintree';
+import { PaypalScriptLoader } from '../payment/strategies/paypal';
+
+import { BraintreePaypalButtonStrategy, CheckoutButtonStrategy } from './strategies';
+
+export default function createCheckoutButtonRegistry(store: CheckoutStore): Registry<CheckoutButtonStrategy> {
+    const registry = new Registry<CheckoutButtonStrategy>();
+    const requestSender = createRequestSender();
+    const checkoutActionCreator = new CheckoutActionCreator(
+        new CheckoutRequestSender(requestSender),
+        new ConfigActionCreator(new ConfigRequestSender(requestSender))
+    );
+
+    registry.register('braintreepaypal', () =>
+        new BraintreePaypalButtonStrategy(
+            store,
+            checkoutActionCreator,
+            new BraintreeSDKCreator(new BraintreeScriptLoader(getScriptLoader())),
+            new PaypalScriptLoader(getScriptLoader()),
+            createFormPoster()
+        )
+    );
+
+    registry.register('braintreepaypalcredit', () =>
+        new BraintreePaypalButtonStrategy(
+            store,
+            checkoutActionCreator,
+            new BraintreeSDKCreator(new BraintreeScriptLoader(getScriptLoader())),
```

Comment:
Is it possible that a store uses both paypal buttons? if we instantiate `BraintreeScriptLoader`, could we be loading the library twice?

Reply:
`getScriptLoader` returns a singleton, which internally manages the caching. But I can move it to a variable so it's clearer.